uk.co.compucorp.civicrm.hrleaveandabsences/tests/phpunit/CRM/HRLeaveAndAbsences/Service/LeaveRequestTest.php

```diff
@@ -187,16 +187,17 @@ public function testCreateThrowsAnExceptionWhenLeaveApproverUpdatesDatesForLeave
   }
 
   /**
-   * @expectedException RuntimeException
-   * @expectedExceptionMessage You are not allowed to change the request dates
+   * @dataProvider openLeaveRequestStatusesDataProvider
    */
-  public function testCreateThrowsAnExceptionWhenAdminUpdatesDatesForLeaveRequest() {
-    $contactID = 5;
-    $params = $this->getDefaultParams(['contact_id' => $contactID]);
+  public function testCreateDoesNotThrowAnExceptionWhenAdminUpdatesDatesForAnyOpenRequest($status) {
```

Comment:
Since we will be having a test in LeaveRequestRights service that already tests that an admin can change dates for a leave request for different statuses, then we don't need to test with various statuses here.